File: Matrix/Lucky Numbers/lucky numbers.java
import java.util.*;

class Solution {
    public List<Integer> luckyNumbers(int[][] matrix) {
        List<Integer> v = new ArrayList<>();

        for (int i = 0; i < matrix.length; i++) {
            int a = matrix[i][0];
            int j = 0;
            for (int col = 1; col < matrix[i].length; col++) {
                if (matrix[i][col] < a) {
                    a = matrix[i][col];
                    j = col;
                }
            }

            int maxi = Integer.MIN_VALUE;
            for (int k = 0; k < matrix.length; k++) {
                maxi = Math.max(maxi, matrix[k][j]);
            }

            if (maxi == a) {
                v.add(a);
            }
        }

        return v;
    }
}

File: Matrix/Lucky Numbers/lucky numbers.cpp
class Solution {
    public:
        vector<int> luckyNumbers(vector<vector<int>>& matrix) {
            vector<int> v;
            for(int i=0;i<matrix.size();i++){
                int a=*std::min_element(matrix[i].begin(),matrix[i].end());
                int j = std::min_element(matrix[i].begin(), matrix[i].end()) - matrix[i].begin();
    
                int maxi=INT_MIN;
                int ans;
                int k=0;
                while(k<matrix.size()){
                    ans=matrix[k][j];
                    maxi=std::max(maxi,ans);
                    k++;
                }
                if(maxi==a){
                    v.push_back(a);
                }
            }
            return v;
        }
    };
    
    
    